client-v2/src/shared/components/input/CreateInput.tsx

```diff
@@ -44,20 +52,59 @@ export default (
     React.HTMLAttributes<HTMLInputElement> & StyledProps<any>
   >,
   type?: string
-) => ({ label, input, originalValue, ...rest }: Props) => (
-  <InputContainer>
-    {label && (
-      <InputLabel htmlFor={label}>
-        {label}
-        {originalValue && input.value !== originalValue && (
-          <RewindButton onClick={() => input.onChange(originalValue)}>
-            <RewindIcon />
-          </RewindButton>
-        )}
-      </InputLabel>
-    )}
-    <InputComponentContainer>
-      <Component id={label} {...input} {...rest} type={type} />
-    </InputComponentContainer>
-  </InputContainer>
-);
+) => {
```

Comment:
This is really neat! Now that our interface depends upon the incoming component being a text field or textarea, it might be better to make this function name more specific -- it wouldn't make sense to supply it with a checkbox input, for example, but there's nothing to suggest that wouldn't be a good idea.

Reply:
It looks like `createInput` is exclusively used by InputText and InputTextArea. As a simple solution, we can rename the file and the function to be more clear about this fact.